doc/guides/test/sanitycheck.rst

```diff
@@ -19,6 +19,26 @@ environment, but it does sufficient testing by building samples and
 tests for different boards and different configurations to help keep the
 full code tree buildable.
 
+sanitycheck's console output doesn't show any difference between a test
+that ran successfully versus a test that was only compiled
+successfully. There are a few reasons why sanitycheck will only build
+a test and not run it:
+
+- The test is marked as ``build_only: true`` in its ``.yaml``
+  configuration files. This is the case for a number of application code
+  examples in the ``samples/`` directory; they can be run but not from
+  sanitycheck.
```

Comment:
that statement is not entirely correct, I would remove the second part of the paragraph, build_only should really be for tests that can't run on any hardware. We have a few samples/tests that are marked as such for various reasons or by mistake.

Reply:
Thanks @nashif 

Below is the list of the 19 `build_only` tests which have some kind of `main` function; that fairly large number 19 made me add that sentence. I'll remove it (and squash @dbkinder's fixup and force push)

Briefly looking at their C code the majority of these look like they are or were meant to be run on some hardware. The small rest have some `ztest_` code.
I can try removing `build_only` for the few that have some `ztest_` code (but not in this commit). So you think all the others are either some mistake and/or missing a `harness`?
```shell
git grep void.*main $(git grep -l build_only -- tests/ samples/ | while read -r; do echo $(dirname "$REPLY"); done)

samples/display/ili9340/src/main.c:void main(void)
samples/sensor/amg88xx/src/main.c:void main(void)
samples/sensor/ms5837/src/main.c:void main(void)
samples/sensor/sht3xd/src/main.c:void main(void)
samples/sensor/vl53l0x/src/main.c:void main(void)
samples/subsys/fs/src/main.c:void main(void)
samples/subsys/ipc/ipm_imx/src/main.c:void main(void)
samples/subsys/ipc/openamp/remote/src/main.c:void main(void)
samples/subsys/ipc/openamp/src/main.c:void main(void)
samples/testing/integration/src/main.c:void test_main(void)
tests/application_development/cpp/src/main.cpp:void test_main(void)
tests/bluetooth/init/src/main.c:void test_main(void)
tests/bluetooth/mesh/src/main.c:void main(void)
tests/boards/native_posix/rtc/src/main.c:void test_main(void)
tests/drivers/build_all/src/main.c:void main(void)
tests/kernel/timer/timer_api/src/main.c:void test_timer_remaining_get(void)
tests/kernel/timer/timer_api/src/main.c:void test_main(void)
tests/net/all/src/main.c:void test_main(void)
tests/unit/bluetooth/at/src/main.c:void test_main(void)
```

Reply:
build_only has nothing to do with ztest, lets not get things confused. Many of samples need to be modified and build_only removed, instead a sensor harness need to be added.